### ISSUE TITLE
`python.linting.ignorePatterns` config value is ignored even if `python.analysis.ignore` is not set

### ISSUE THREAD

**Issue by debonte** [microsoft/pylance-release#4100]
## Environment data
-   Language Server version: 2023.3.23

### test.py
```python
x: int = "str"
```

### settings.json
```json
"python.linting.ignorePatterns": ["test.py"]
```

## Repro Steps
1. Create Python file that generates any diagnostics. Example above.
2. Ignore that file via `python.linting.ignorePatterns` in settings.json. Ensure `python.analysis.ignore` is not set.

## Expected behavior

No diagnostics reported for Python file.

## Actual behavior

Diagnostics. Looks like the problem is that `python.analysis.ignore` defaults to an empty list, so the following assignment does not cause `serverSettings.ignoreFileSpecs` to be set to the `python.linting.ignorePatterns` value as expected:

```typescript
serverSettings.ignoreFileSpecs = tryGetStringValues(
    pythonAnalysisSection.ignore ?? pythonSection?.linting?.ignorePatterns
);
```

**Comment by erictraut** [comment]
Why would pylance pay attention to the setting `python.linting.ignorePatterns`? Shouldn't this setting be reserved for linters like pylint?

**Comment by Spitfire1900** [comment]
@erictraut
![image](https://user-images.githubusercontent.com/6462596/225991908-aac27692-9b0c-4b72-bcd8-b8d890f757ac.png)

**Comment by Spitfire1900** [comment]
I tried the following workaround to no avail: `"python.analysis.ignore": null`

**Comment by erictraut** [comment]
My point is that I don't think that Pylance should pay any attention to the `python.linting.ignorePatterns` setting under any conditions. It's a setting that is not applicable to Pylance. It's designed for use with other tools — specifically, linters like pylint or flake8. Pylance should honor the setting `python.analysis.ignore`, which accepts an array of globs patterns for files or directories that you would like it to ignore.

**Comment by debonte** [comment]
@heejaechang, what was the argument for supporting `python.linting.ignorePatterns` in Pylance?

**Comment by heejaechang** [comment]
we had a bug about supporting ignorePatterns - https://github.com/microsoft/pylance-release/issues/3261

we can discuss whether we should use `ignorePatterns` or not, but our previous decision was using it (since we accepted the request as `enhencement`)

and regardless how we decide, it looks like there is a bug how we check both ignored and ignorePatterns settings.

**Comment by erictraut** [comment]
My recommendation is to consider removing support for `python.linter.ignorePatterns` in pylance. IMO, this is not a setting that pylance should honor. It is intended for linters, not type analyzers. The `python.analysis.ignore` is the setting that should be used for pylance.

**Comment by heejaechang** [comment]
tagging @judej @luabud for decision.

**Comment by luabud** [comment]
@karthiknadig can correct me if I'm wrong, but I think the plan is to deprecate the `python.linting.ignorePatterns` namespace once there's no longer built-in support for linters in the Python extension. So it might be better to remove support for it in Pylance. 
I think there was a similar discussion happening around another setting on the Python extension side, but I don't recall what it was. Do you remember, Karthik?

**Comment by karthiknadig** [comment]
> the plan is to deprecate the python.linting.ignorePatterns namespace once there's no longer built-in support for linters in the Python extension.

Yes, once linters are fully moved to their own extensions we plan on removing the support for the `ignorePatterns` in the Python extension.

**Comment by Spitfire1900** [comment]
In practice why would the to settings ever be set to different values?

**Comment by erictraut** [comment]
Because linters and type checkers are two different classes of tools that have different purposes. There is some overlap, but they're quite distinct. The settings for one tool should not impact the settings for another tool. If you happen to want to set the same settings for both, you should be able to do this, but you shouldn't be forced to have the same settings for both.

**Comment by debonte** [comment]
This issue has been fixed in prerelease version 2023.3.31, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023331-22-March-2023-PreRelease)
